File: done/InvertBinaryTree.java
import pack.TreeNode;

class InvertBinaryTree {
  public TreeNode invertTree(TreeNode root) {
    if ( root == null) return null;
    return parseSubTree(root);
  }

  TreeNode parseSubTree(TreeNode root) {
    TreeNode output = new TreeNode(root.val);
    if( root.right != null) output.left = parseSubTree(root.right);
    if( root.left != null) output.right = parseSubTree(root.left);
    return output;
  }

  public static void main(String args[]){ 
    Object[] input = {4, 2, 7, 1, 3, 6, 9};
    TreeNode t = new TreeNode().populateTree(input);
    t.printTree();

    new InvertBinaryTree().invertTree(t).printTree();
  }  
}

File: done/InvertBinaryTree.py
from timeit import default_timer as timer
from typing import List, Optional
from pack.TreeNode import TreeNode

class Solution:
    def __init__(self, root):
        self.root = root
        print("init")

    def invertTree(self, root: Optional[TreeNode], invr = None) -> Optional[TreeNode]:
        if root == None: return root
        invr = TreeNode(root.val)
        invr.left = self.invertTree(root.right, invr.left)
        invr.right = self.invertTree(root.left, invr.right)

        return invr

    def main(self):
        invr = self.invertTree(self.root)
        invr.traverse()
        return

if __name__ == "__main__":
    root = TreeNode()
    root.populateFromScratch(16)
    root.traverse()

    # root = None
    sol = Solution(root)
    start = timer()
    sol.main()
    print(f"{timer() - start:.20f}")
